helix-term/src/commands.rs

```diff
@@ -3016,12 +3154,17 @@ fn replace_with_yanked(cx: &mut Context) {
 
     if let Some(values) = registers.read(reg_name) {
         if let Some(yank) = values.first() {
-            let transaction =
-                Transaction::change_by_selection(doc.text(), doc.selection(view.id), |range| {
-                    let max_to = rope_end_without_line_ending(&doc.text().slice(..));
-                    let to = std::cmp::min(max_to, range.to() + 1);
-                    (range.from(), to, Some(yank.as_str().into()))
-                });
+            let selection = doc
+                .selection(view.id)
+                .clone()
+                .min_width_1(doc.text().slice(..));
+            let transaction = Transaction::change_by_selection(doc.text(), &selection, |range| {
+                if !range.is_empty() {
```

Comment:
I wonder, rather than check for empty ranges here, how about just filtering them out before passing them into change_by_selection?

Reply:
Ah!  Good call.

Reply:
In attempting to implement this, it turns out that it's actually a bigger change.  To use iterator adaptors like `.filter()` we'll need to pass an iterator rather than a `&Selection` to `Transaction::change_by_selection()`, which it doesn't current accept.  I think that's a good change anyway, but I'd rather do that as a separate PR after this one.